web/src/main/java/se/inera/intyg/webcert/web/service/facade/list/ResourceLinkListHelperImpl.java

```diff
@@ -0,0 +1,175 @@
+                Personnummer.createPersonnummer(entry.getCivicRegistrationNumber()).get(),
+                entry.isTestIndicator()
+        );
+
+        if (certificateAccessServiceHelper.isAllowToRead(accessEvaluationParameters)) {
+            links.add(new ActionLink(ActionLinkType.LASA_INTYG));
+        }
+
+        if (certificateAccessServiceHelper.isAllowToRenew(accessEvaluationParameters)) {
+            links.add(new ActionLink(ActionLinkType.FORNYA_INTYG));
+        }
+
+        return links;
+    }
+
+    private List<ResourceLinkDTO> convertResourceLinks(List<ActionLink> links, CertificateListItemStatus status) {
+        return links.stream()
+                .map((link) -> getConvertedResourceLink(link, status))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private List<ResourceLinkDTO> convertResourceLinks(
+                List<ActionLink> links, String unitId, String certificateType,
+                CertificateListItemStatus status, CertificateRelations relations
+            ) {
+        return links.stream()
+                .map((link) -> getConvertedResourceLink(link, unitId, certificateType, status, relations))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private ResourceLinkDTO getConvertedResourceLink(ActionLink link, CertificateListItemStatus status) {
+        if (link.getType() == ActionLinkType.LASA_INTYG) {
+            return ResourceLinkFactory.read();
+        } else if (validateForward(link, getCertificateStatus(status))) {
+            return CertificateForwardFunction.createGenericResourceLink();
+        }
+        return null;
+    }
+
+    private CertificateStatus getCertificateStatus(CertificateListItemStatus status) {
+        if (status == CertificateListItemStatus.LOCKED) {
+            return CertificateStatus.LOCKED;
+        } else if (status == CertificateListItemStatus.INCOMPLETE || status == CertificateListItemStatus.COMPLETE) {
+            return CertificateStatus.UNSIGNED;
+        } else if (status == CertificateListItemStatus.REVOKED) {
+            return CertificateStatus.REVOKED;
+        } else {
+            return CertificateStatus.SIGNED;
+        }
+    }
+
+    private boolean validateForward(ActionLink link, CertificateStatus status) {
+        return link.getType() == ActionLinkType.VIDAREBEFORDRA_UTKAST && CertificateForwardFunction.validate(status);
+    }
+
+    private ResourceLinkDTO getConvertedResourceLink(
+                ActionLink link, String savedUnitId, String certificateType,
+                CertificateListItemStatus status, CertificateRelations relations
+            ) {
+        final var convertedResourceLink = getConvertedResourceLink(link, status);
+        if (convertedResourceLink != null) {
+            return convertedResourceLink;
+        } else if (validateRenew(link, certificateType, relations, getCertificateStatus(status))) {
```

Comment:
Little tricky code to read... but you could remove the else and just do an if. Could make it little more readable.